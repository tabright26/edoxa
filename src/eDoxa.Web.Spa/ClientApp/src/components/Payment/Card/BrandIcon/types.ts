import { SizeProp } from "@fortawesome/fontawesome-svg-core";

export type BrandProp = "visa" | "amex" | "mastercard" | "discover";

export interface CardBrandIconProps {
  className: string;
  brand: BrandProp;
  size?: SizeProp;
}
