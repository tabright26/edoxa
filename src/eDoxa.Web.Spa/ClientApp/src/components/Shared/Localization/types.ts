export interface Country {
  readonly name: string;
  readonly twoDigitIso: string;
}

export interface Address {
  country: Country;
  line1: string;
  line2: string | null;
  city: string;
  state: string | null;
  postalCode: string;
}
