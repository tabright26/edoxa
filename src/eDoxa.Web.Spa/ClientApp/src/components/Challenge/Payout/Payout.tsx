import React, { FunctionComponent } from "react";
import { Table, Card, CardHeader } from "reactstrap";
import Format from "components/Shared/Format";
import { ChallengeId, ChallengePayout } from "types";

interface Props {
  readonly challengeId: ChallengeId;
  readonly payout: ChallengePayout;
}

const ArenaChallengePayout: FunctionComponent<Props> = ({ payout }) => {
  let prevSize = 1;
  let nextSize = 0;
  return (
    <Card className="text-center">
      <CardHeader className="bg-gray-900">
        <strong className="text-uppercase">Payout</strong>
      </CardHeader>
      <Table striped bordered hover size="sm" variant="dark" className="m-0">
        <thead>
          <tr>
            <th className="align-middle w-50">Position</th>
            <th className="align-middle w-50">Prize</th>
          </tr>
        </thead>
        <tbody>
          {payout.buckets.map((bucket, index) => {
            const bucketSize = bucket.size;
            nextSize += bucketSize;
            const bucketRender = (
              <tr key={index}>
                <td className="align-middle w-50">
                  {
                    <Format.PayoutBucket
                      prevSize={prevSize}
                      nextSize={nextSize}
                    />
                  }
                </td>
                <td className="align-middle w-50">
                  {
                    <Format.Currency
                      alignment="center"
                      currency={payout.prizePool.currency}
                      amount={bucket.prize}
                    />
                  }
                </td>
              </tr>
            );
            prevSize += bucketSize;
            return bucketRender;
          })}
        </tbody>
      </Table>
    </Card>
  );
};

export default ArenaChallengePayout;
