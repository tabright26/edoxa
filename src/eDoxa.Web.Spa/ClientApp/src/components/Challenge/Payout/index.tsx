import React, { FunctionComponent } from "react";
import { Table, Card, CardHeader } from "reactstrap";
import Format from "components/Shared/Format";
import { ChallengeId, ChallengePayout } from "types";
import { connect, MapStateToProps } from "react-redux";
import { RootState } from "store/types";
import { RouteComponentProps, withRouter } from "react-router-dom";
import { compose } from "recompose";

type Params = {
  readonly challengeId?: ChallengeId;
};

type OwnProps = RouteComponentProps<Params> & Params;

type StateProps = {
  readonly payout: ChallengePayout;
};

type InnerProps = OwnProps & StateProps;

type OutterProps = Params;

type Props = InnerProps & OutterProps;

const Payout: FunctionComponent<Props> = ({ payout }) => {
  let prevSize = 1;
  let nextSize = 0;
  return (
    <Card className="text-center">
      <CardHeader className="bg-gray-900">
        <strong className="text-uppercase">Payout</strong>
      </CardHeader>
      <Table striped bordered hover size="sm" variant="dark" className="m-0">
        <thead>
          <tr>
            <th className="align-middle w-50">Position</th>
            <th className="align-middle w-50">Prize</th>
          </tr>
        </thead>
        <tbody>
          {payout.buckets.map((bucket, index) => {
            const bucketSize = bucket.size;
            nextSize += bucketSize;
            const bucketRender = (
              <tr key={index}>
                <td className="align-middle w-50">
                  <Format.PayoutBucket
                    prevSize={prevSize}
                    nextSize={nextSize}
                  />
                </td>
                <td className="align-middle w-50">
                  <Format.Currency
                    alignment="center"
                    currency={payout.prizePool.currency}
                    amount={bucket.prize}
                  />
                </td>
              </tr>
            );
            prevSize += bucketSize;
            return bucketRender;
          })}
        </tbody>
      </Table>
    </Card>
  );
};

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (
  state,
  ownProps
) => {
  const { data } = state.root.challenge;
  const challenge = data.find(
    challenge =>
      challenge.id ===
      (ownProps.match
        ? ownProps.match.params.challengeId
        : ownProps.challengeId)
  );
  return {
    payout: challenge.payout
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withRouter,
  connect(mapStateToProps)
);

export default enhance(Payout);
