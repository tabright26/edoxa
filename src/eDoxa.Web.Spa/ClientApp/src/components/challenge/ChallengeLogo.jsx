import React from 'react';
import { Card } from 'react-bootstrap';

import logo from '../../assets/images/logos/LeagueOfLegends.png';

const ChallengeLogo = () => (
  <Card className="bg-dark mx-3 my-4">
    <Card.Img className="p-4" src={logo} />
  </Card>
);

export default ChallengeLogo;
