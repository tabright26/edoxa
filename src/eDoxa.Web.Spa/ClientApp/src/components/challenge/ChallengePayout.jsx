import React from 'react';
import { Table, Card } from 'react-bootstrap';

import Spinner from '../Shared/Spinner';
import CurrencyFormat from '../Shared/Formaters/CurrencyFormat';
import BucketFormat from '../Shared/Formaters/BucketFormat';

const Render = ({ challenge }) => {
  if (!challenge) {
    return (
      <Card.Body>
        <Spinner />
      </Card.Body>
    );
  } else {
    let prevSize = 1;
    let nextSize = 0;
    return (
      <Table striped bordered hover size="sm" variant="dark" className="m-0">
        <thead>
          <tr>
            <th>Position</th>
            <th>Prize</th>
          </tr>
        </thead>
        <tbody>
          {/* {challenge.payout.buckets.map((bucket, index) => {
            const bucketSize = bucket.size;
            nextSize += bucketSize;
            const bucketRender = (
              <tr key={index}>
                <td>
                  { <BucketFormat prevSize={prevSize} nextSize={nextSize} /> }
                </td>
                <td>
                  { <CurrencyFormat
                    currency={challenge.payout.prizePool.currency}
                    amount={bucket.prize}
                  /> }
                </td>
              </tr>
            );
            prevSize += bucketSize;
            return bucketRender;
          })} */}
        </tbody>
      </Table>
    );
  }
};

const ChallengePayout = ({ challenge }) => {
  return (
    <Card bg="dark" className="my-4 text-light text-center">
      <Card.Header as="h5">
        {/* Payout ({challenge ? challenge.payout.prizePool.currency : ''}) */}
      </Card.Header>
      <Render challenge={challenge} />
    </Card>
  );
};

export default ChallengePayout;
