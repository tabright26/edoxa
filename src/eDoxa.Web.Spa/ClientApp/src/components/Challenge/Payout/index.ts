import ChallengePayout from "./Container";

export default ChallengePayout;
