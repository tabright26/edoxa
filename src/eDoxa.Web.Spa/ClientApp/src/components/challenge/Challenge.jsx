import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';

import ChallengeLogo from './ChallengeLogo';
import ChallengeSetup from './ChallengeSetup';
import ChallengeScoreboard from './ChallengeScoreboard';
import ChallengeScoring from './ChallengeScoring';
import ChallengePayout from './ChallengePayout';

import faker from 'faker';

import { findChallenge } from '../../store/actions/arenaChallengeActions';

faker.seed(1);

class Challenge extends Component {
  componentDidMount() {
    // Check if the challenge exists in the store.
    if (!this.props.challenge) {
      // If the challenge does not exist, dispatch a redux action to find the challenge with the API call.
      // This call will update the store with the HTTP response call.
      this.props.actions.findChallenge();
    }
  }

  render() {
    return (
      <Container fluid={true}>
        <Row>
          <ChallengeLogo />
          <Col xs={4} className="ml-4">
            <ChallengeSetup challenge={this.props.challenge} />
          </Col>
        </Row>
        <Row>
          <Col xs={10}>
            <ChallengeScoreboard challenge={this.props.challenge} />
          </Col>
          <Col xs={2}>
            <Row>
              <Col xs={12}>
                <ChallengeScoring challenge={this.props.challenge} />
              </Col>
              <Col xs={12}>
                <ChallengePayout challenge={this.props.challenge} />
              </Col>
            </Row>
          </Col>
        </Row>
      </Container>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    challenge: state.arena.challenges.find(
      challenge => challenge.id === ownProps.match.params.challengeId
    )
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    actions: {
      findChallenge: () =>
        dispatch(findChallenge(ownProps.match.params.challengeId))
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Challenge);
