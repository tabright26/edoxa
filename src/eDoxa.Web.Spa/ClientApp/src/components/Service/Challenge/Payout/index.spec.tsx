import React from "react";
import renderer from "react-test-renderer";
import Payout from ".";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import store from "store";
import { createChallenge } from "test/service/challenges";
import { createAction } from "test/helper";
import { LOAD_CHALLENGE_SUCCESS } from "store/actions/challenge/types";

it("renders without crashing", () => {
  const challenge = createChallenge();
  const action = createAction(LOAD_CHALLENGE_SUCCESS, challenge);
  store.dispatch(action);

  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Payout challengeId={challenge.id} />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
