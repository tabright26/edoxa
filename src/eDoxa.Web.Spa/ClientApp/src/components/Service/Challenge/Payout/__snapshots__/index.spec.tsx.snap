// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="text-center card"
>
  <div
    className="card-header"
  >
    <strong
      className="text-uppercase"
    >
      Payout
    </strong>
  </div>
  <table
    className="m-0 border-0 table table-sm table-bordered table-striped table-hover"
    color="dark"
  >
    <thead>
      <tr>
        <th
          className="align-middle w-50"
        >
          Position
        </th>
        <th
          className="align-middle w-50"
        >
          Prize
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
